common-service-core/src/main/java/com/ca/mfaas/message/core/Message.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.message.core;
+
+import com.ca.mfaas.message.api.ApiMessage;
+import com.ca.mfaas.message.api.ApiMessageView;
+import com.ca.mfaas.message.template.MessageTemplate;
+import org.apache.commons.lang.StringEscapeUtils;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Message creator immutable class
+ */
+public final class Message {
+
+    public static final String INVALID_KEY_MESSAGE = "com.ca.mfaas.common.invalidMessageKey";
+    public static final String INVALID_MESSAGE_TEXT_FORMAT = "com.ca.mfaas.common.invalidMessageTextFormat";
+
+    private final String requestedKey;
+    private final MessageTemplate messageTemplate;
+    private final Object[] messageParameters;
+
+
+    private Message(String requestedKey,
+                    MessageTemplate messageTemplate,
+                    Object[] messageParameters) {
+        this.requestedKey = requestedKey;
+        this.messageTemplate = messageTemplate;
+        this.messageParameters = messageParameters;
+    }
+
+    /**
+     * Return a {@link Message} object for the specified key after text and parameters validation.
```

Comment:
Is that the question to this statement "does it return...?" If yes, then "Returns a {@link Message} object..."

Reply:
same here, returns a ...

Reply:
Cool!